### DISCUSSION TITLE
ユーザーランキングや表彰者の掲示は無くてもいいのでは？

### DISCUSSION THREAD

**Discussion by ghost** [increments/qiita-discussions#194] [要望・アイデア・不具合]
2022/3/31追記：表現が強すぎたので修正しました

2022/4/1 より Contribution の計算式が見直されますが、そもそも能動的なアクションで Contribution が加算される時点で

- 内容が薄い記事を投稿しまくる
- 他人の記事に簡単な編集リクエストを送りまくる

といった行為を行うことで Contribution を稼ぎ、ユーザーランキングを上り詰めて表彰までしてもらうことができます。  
私は今ここに違和感を抱いています。

しかし利用規約に違反しない限り、記事投稿や編集リクエスト自体を Contribution とみなしたいという意図は理解できます。  
「かけた手間に対して Contribution が見合っていない」という意見もあるとは思いますが、私はそれはちょっと違うかなあと思っています。最終的に誰かの役に立ったなら、手間や時間をどれだけかけたかは重要ではないんじゃないかな、と……。

自分が感じている違和感をよく考えた結果、行っている行為自体は別にいいのですが、「ユーザーランキングにのることや表彰されることを目的として制度の穴を突く」というのがあまり良くないように感じるという結論に至りました。  
（＝もし Qiita にユーザーランキングが無く、表彰も Qiita とユーザーの間で閉じたものであるとしたら、誰がどんな目的で Contribution を稼いでいてもどうでもいいです。）

そして「ランキング」がある以上そういう人は現れるよな、と思ったのですが、今まで数年間にわたり Qiita を使ってきてユーザーランキングがあってよかったと思ったことが一度もありません。  
表彰者の掲載も別に無くていいです。

つまり私にとってユーザーランキングや表彰者の掲載は百害あって一利なしなのですが、それらは Qiita が目指す世界にどのように貢献するのでしょうか？
  
スポーツの世界などではランキングで競い合い切磋琢磨するほうが良い結果を生むのでしょうが、エンジニアコミュニティで「Contribution ランキングで一位を取ろう！」と思って再利用性・汎用性がある素晴らしい記事を書こうとする人はあまりいないと思います。  
正直なところマトモな方法でランキング入りを狙うのって凡人には無理だと思うので目指そうとも思いません。 

表彰についても、ただ Qiita としてユーザーに感謝したいのならトロフィーを送るだけでいいのではないでしょうか？  
もしくはその方のユーザーページへ何か表示してもいいと思いますが、わざわざみんなに見えるところに表彰者を掲載しなくてもいいと思います。

ちょっと纏まりがない文章なのですが、以上のような考えから、ユーザーランキングや表彰者の表示を無くすのはどうか？　というアイデアを提案します。

**Comment by kskwtnk** [comment]
@asa2200 
ご意見ありがとうございます！
ここまで詳細に、色々な文脈も含めて考えてくださって大変ありがたいです 🙏 

まず、我々が目指している世界としては以下のようなポイントがあります（[QiitaのAboutページ](https://qiita.com/about)に記載の内容です）

> Qiita上で発信や評価などの活動をすればするほど、あなたに合った記事が届き、ほかのエンジニアと繋がり、エンジニアとして成長していける場所を目指しています。

そして、Qiitaで行動をしてくれた方の貢献度を可視化し、公表することも上記の文脈に含まれると考えています
ランキングや表彰プログラムを用意しているのも、こういった目的です

ただ、あくまでここまでの話は「運営としての考え」がメインでした
仰る通りトロフィーを個別に贈るだけの方が良いのかもしれません
ユーザーの皆さんにとって、あるいはコミュニティ全体にとって、より嬉しいコミュニケーションは色々あると思います

今回のいただいたご意見も受け止め、今私たちが考えていることをより良いかたちで皆さまにお届けできるように改善していきます！
（もちろん、このDiscussionを見ている他の方からの意見もお待ちしています！）

**Reply by ghost** [reply]
@xrxoxcxox 
ご回答ありがとうございます。

> > Qiita上で発信や評価などの活動をすればするほど、あなたに合った記事が届き、ほかのエンジニアと繋がり、エンジニアとして成長していける場所を目指しています。 
>  
> Qiitaで行動をしてくれた方の貢献度を可視化し、公表することも上記の文脈に含まれると考えています

なるほど……。  
正直なところ、どういう論理でそうなるのかがよくわかりませんでした。

現在の計算式では、Contribution の値から判断できることって純粋に「Qiita で活動をした」しかありません。  
そのため「ほかのエンジニアと繋がり」の文脈で言うと、繋がりたい相手を判断することにはあまり使えないと思います。  
（私は 100 件の薄い記事を投稿したユーザーより、50 LGTM つく記事を 1 件投稿したユーザーと繋がりたいです。）


「エンジニアとして成長していける」の文脈だとすれば、Qiita としてはそもそも Qiita での活動＝成長に繋がると考えていらっしゃった？  
そしてランキング入りや表彰を活動のモチベーションにしている人が意外とたくさんいて（？）、

ランキングや表彰で Contribution が高いユーザーを称賛する  
＝ユーザーが Qiita で活動するモチベーション向上に寄与する  
＝成長に繋がる

ということですかね？

そうだとすれば、ランキング入りや表彰をモチベーションとして記事を投稿しまくったり、編集リクエストを送りまくったりする人が現れるのは狙い通りと言えますよね。  
ただしその結果が成長に繋がらないのは、本来の目的を達成できてないので問題と言えます。

でも何らかの行動がその人の成長に繋がっているかどうかなんて、客観的には計測のしようがありません。

とにかく「コミュニティへの貢献」に繋がっていれば成長だと認めるよ、ということだとしても、今の Contribution は「こんなのは我々のコミュニティへの貢献ではない！」というコミュニティの感情を反映できないという点において不備があります。  
「不公平感」という感情を理由に編集リクエストの貢献度を下げたわけなので、たぶんコミュニティの感情はユーザーランキングに反映されるべきなのですよね？  
まあ以前もポエムがバズっていいねを集める現象はありましたが、それでも「俺はポエムは貢献じゃないと思う」と思ったら「いいねを押さない」という意思表明ができました。  
でも能動的アクションを貢献と認められると、もう意思の介在の余地がない……。

たぶん今のユーザーランキングや表彰は本来の目的を達成できないので、やっぱり不要ではないでしょうか……？

---

あとさっき気づいたんですが、Qiita 表彰プログラムのページに

https://qiita.com/qiita-award

> Qiita 表彰プログラムは、多くの人に記事を読まれ、リアクションを集めている皆さまのためのプログラムです。

と書いてありますけど、もはや今は違いますよね。Contribution の変更に伴う修正が漏れているかもです。

**Reply by radian-jp** [reply]
真に記事の質が反映されるようなランキングならあってもいいとは思いますが、現状抜け穴が多すぎますね。
記事の質の向上に繋がっているどころか、むしろ逆効果になっているとすら感じます。
編集リクエストでContribution上がるのは、正直無くていいと思っています。
もしくは、Contributionのカテゴリ分けをした方がよいでしょう。
記事で活躍している人、質問の回答で活躍している人、みたいに区別付きますし。

> 行動指針 ‐ Qiitaの目指す世界
> エンジニアにとって再利用性・汎用性の高い情報が集まる場をつくろう

であれば、LGMTよりストックを重視すべきかなと。
ストックは見返す可能性があるからストックする訳で、再利用性が高い記事だと判断出来るのではないかと。
（LGMT多い記事は概ねストックも多いですが、ポエム的なもので稼いでる場合に若干評価下げれると思うので）

**Reply by ghost** [reply]
@radian-jp 

> もしくは、Contributionのカテゴリ分けをした方がよいでしょう。  
> 記事で活躍している人、質問の回答で活躍している人、みたいに区別付きますし。

わかります。  
カテゴリごとのランキングなら得られるものはありそうです。総合ランキングは本当に得られる情報が乏しすぎます……。

> 行動指針 ‐ Qiitaの目指す世界  
> エンジニアにとって再利用性・汎用性の高い情報が集まる場をつくろう

私も今の Contribution はこの理想には寄与しないと思います。  
しかも、今「再利用性・汎用性の高い情報が集まる場」の実現に役立つドンピシャの評価システムは無いように思われますね……。

私はストックも若干難しいと思っていて、ストック機能は  
https://help.qiita.com/ja/articles/qiita-stocks  
> ストック機能とは、あとでじっくり読みたい、繰り返し読みたい記事を保存しておくことができる機能です。  

ということで「あとで読む」的な使われ方も想定しているようなのです。  
つまり時間が無くて読めなかった記事をストックしておき、読んだらストックから外してしまう方が一定数いる可能性があります。  

実際、「いいね」導入時のブログによると  
https://blog.qiita.com/153200849029-2/  

「あとで読む」として使われうるストックは記事への評価に転用しないべきだというのが当初の想定だったようです。

でも LGTM も結局は「いいね」でしかないので微妙ですよね……。

**Comment by kskwtnk** [comment]
@asa2200 
詳細にありがとうございます！

回答するより先になってしまって申し訳ありませんが、1度オンラインでお話させていただけたら……と率直に思っています 🙏 

これだけ細かくというか、Contributionの全体的な構造についてご意見をいただけるのが非常に希少です
（個別の要素についてのご意見をいただくことはあるのですが）

そのため以下のようなディスカッションができたら非常に嬉しいです

- 運営として思っている内容が、実際ユーザーの皆さまからどう見えているのか
- 今のやり方がベストではないと思うので、他にはどんなものがありそうか
    - いきなり全部をなくすというのがあまり現実的ではないのもあり、どんな段階を踏んでいけそうか

もしよろしければ別途ご連絡させていただきたいのですが、お願いしてもよろしいでしょうか？
（GitHubにメールアドレスのご登録がなかったので、他の手段を提示させていただきます）

**Reply by ghost** [reply]
@xrxoxcxox 私とディスカッションするより Qiita ユーザーへ全体アンケートをするほうが有意義なデータが得られるのでは……？という気がしていますが、一応私の GitHub のユーザーページに Google フォームの URL をセットしておきましたので、もし必要であればそちらからご連絡ください。  
メールにて返信いたします。  
（私から xrxoxcxox さんへメールでご連絡しようかと思ったのですが、メールの送り主が自分であると証明する手段がなかったため、お手数ですがよろしくお願いいたします）

**Reply by kskwtnk** [reply]
@asa2200 ありがとうございます！この後ご連絡いたします！
